packages/gestalt/src/Button.js

```diff
@@ -76,6 +77,7 @@ export default function Button(props: Props): Element<'button'> {
     [styles.enabled]: !disabled,
     [styles.inline]: inline,
     [styles.block]: !inline,
+    [styles.darkModeGray]: isDarkMode && color === 'gray',
```

Comment:
This needs to be part of the same logic as `colorClass` so it gets handled correctly with the disabled/selected logic. You can see the issue in the docs where in dark mode gray selected has a hover color (incorrect) but red selected does not (correct)